src/Components/NavBar/Menus/NewMobileNavMenu.tsx

```diff
@@ -0,0 +1,197 @@
+import { Box, ChevronIcon, color, Flex, Sans, Separator } from "@artsy/palette"
+import { useSystemContext } from "Artsy"
+import React from "react"
+import styled from "styled-components"
+import { MenuData, MenuLinkData } from "../menuData"
+import { MobileLink } from "./MobileLink"
+import {
+  NavigatorContextProvider,
+  useNavigation,
+} from "./NavigatorContextProvider"
+
+interface Props {
+  isOpen: boolean
+  menuData: MenuData
+}
+
+export const NewMobileNavMenu: React.FC<Props> = props => {
+  const {
+    links: [artworks, artists],
+  } = props.menuData
+
+  const { user } = useSystemContext()
+
+  return (
+    <NavigatorContextProvider>
+      <MenuViewport>
+        <AnimatingMenuWrapper isOpen={props.isOpen}>
+          <ul>
+            <MobileSubmenuLink menu={(artworks as MenuLinkData).menu}>
```

Comment:
I'm not a TypeScript guru, but would it be possible to type these once (like on line 19) vs. having to repeat?

Reply:
I'll give this a try.

Reply:
@sweir27 you were thinking something like:  `[(artworks as MenuLinkData), (artists as MenuLinkData)]`? TypeScript doesn't seem to like that.

@damassi do you know of a better way to type these?